### ISSUE TITLE
[prometheus] server extraArgs does not support flag args

### ISSUE THREAD

**Issue by haizaar** [helm/charts#2626]
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

BUG REPORT

**Version of Helm and Kubernetes**:
Helm 2.7.0, k8s 1.7.8 on GKE

**Which chart**:
prometheus

**What happened**:
Unable to specify flag args with `Values.server.extraArgs`. Particularly, in prometheus 2.x there lots of flag values, e.g. `--web.enable-lifecycle` that work as flag-only, i.e. `--web.enable-lifecycle=1` will not work. However, `server-deployment.yaml` template parses `extraArgs` as key/value and tries to set args in form `--<key>=<value>`. 

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

**Comment by ptagr** [comment]
@haizaar Please use this syntax to specify the `extraArgs`

```
  extraArgs:
    storage.local.retention: 720h
    storage.local.num-fingerprint-mutexes: 4096
```

**Comment by haizaar** [comment]
Perhaps my description was not clear enough. You example converts to 
```
--storage.local.retention=720h --storage.local.num-fingerprint-mutexes=4096
```

However, how how do I specify:
```
--web.enable-lifecycle
```
which is a flag option, i.e. does not accept any argument?

**Comment by tanucojp** [comment]
Hi, it seems this issue is not solved in latest 7.4.5 chart.
